server/src/com/thoughtworks/go/server/service/AgentService.java

```diff
@@ -368,6 +374,17 @@ public String assignCookie(AgentIdentifier identifier) {
         return cookie;
     }
 
+    public Agent findAgentObjectByUuid(String uuid) {
+        Agent agent = agentDao.agentByUuid(uuid);
```

Comment:
Given the original logic was to always lookup in config, was wondering if we should lookup the config by default and if not found, make a trip to db. Does that make sense to you?